src/modules/decklink/producer/decklink_producer.cpp

```diff
@@ -122,10 +122,6 @@ struct Filter
                 }
             }
 
-            if (format_desc.field_count == 2) {
-                filter_spec += ",bwdif=mode=send_field:parity=auto:deint=all";
```

Comment:
What about this?

Reply:
It doesn't seem to do anything. It deinterlaces when the output format is deinterlaced, but since the pipeline is now fully progressive, correct? If you notice issues with it, I can put it back...

Reply:
This will still be needed when the input is interlaced. Before it was checking the output format as the input and output were guaranteed to be the same, but it should really have been checking the input format.

It will need to be tweaked though, as it will need to be always enabled to be able to handle format changes. It might be possible to change the end to `deint=interlaced` to ensure we dont waste cpu attempting to deinterlace a progressive source

Reply:
I dont think that deint=interlace is working correctly in this case as I think it relies on frame metadata which might not be availible in this case as SDI does not support anything like this.

But give it a try.

But I think the bwdif filter needs to be set to deint=all for interlaced input formats and skipped at all for progressive input formats.
But it should never produce bad quality on progressive input even if set to deint=all for any input format as the internal motion estimation skips the deinterlacing when there is nothing to do. Of course the motion estimation does cost some CPU time...

Reply:
@premultiply we are having to convert the decklink frame data into a frame that can be fed into ffmpeg, so it should be easily possible for us to set the metadata of those frames as we do so